repositories/heartbeat.go

```diff
@@ -231,22 +233,29 @@ func (r *HeartbeatRepository) GetUserProjectStats(user *models.User, from, to ti
 	// see https://github.com/muety/wakapi/issues/524#issuecomment-1731668391
 
 	// multi-line string with backticks yields an error with the github.com/glebarez/sqlite driver
+
+	limitClause := "limit ?"
+
+	if r.config.Db.IsMssql() {
+		limitClause = "ROWS fetch next ? rows only"
+	}
+
 	if err := r.db.
 		Raw("with projects as ( "+
-			"select project, user_id, min(time) as first, max(time) as last, count(*) as cnt "+
+			"select project as p, user_id, min(time) as first, max(time) as last, count(*) as cnt "+
 			"from heartbeats "+
 			"where user_id = ? and project != '' "+
 			"and time between ? and ? "+
 			"and language is not null and language != '' and project != '' "+
 			"group by project, user_id "+
 			"order by last desc "+
-			"limit ? offset ? "+
+			"offset ? "+limitClause+
```

Comment:
Getting an error when calling an endpoint that calls down to this function.

```
2024-01-12T15:34:52.198713355+01:00 [ERROR] SQL logic error: near "offset": syntax error (1)
```

Reply:
It seems because SQLite has to put `limit` before `offset`. I am fixing it so the new clause only applies to mssql.